rubix-spi/src/main/java/com/qubole/rubix/spi/BookKeeperFactory.java

```diff
@@ -20,13 +24,84 @@
 import org.apache.thrift.transport.TTransport;
 import org.apache.thrift.transport.TTransportException;
 
+import java.net.SocketException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
 /**
  * Created by sakshia on 5/10/16.
  */
 public class BookKeeperFactory
 {
   BookKeeperService.Iface bookKeeper;
   private static Log log = LogFactory.getLog(BookKeeperFactory.class.getName());
+  private static final AtomicInteger count = new AtomicInteger();
+  private static final AtomicBoolean flag = new AtomicBoolean();
+  private static PoolConfig poolConfig;
+  private static ConcurrentHashMap<String, Integer> concurrentHashMap = new ConcurrentHashMap<>();
+  private static ObjectFactory<TSocket> factory;
+  static ObjectPool pool;
+
+  private static final String LOCALHOST = "localhost";
+
+  static {
+    // TODO: add config.properties file to set this config
+    poolConfig = new PoolConfig();
+    poolConfig.setPartitionSize(10);
+    poolConfig.setMaxSize(2000);
+    poolConfig.setMinSize(100);
+    poolConfig.setDelta(100);
+    poolConfig.setMaxIdleMilliseconds(60 * 1000 * 5);
+    final int socketTimeout = 60000;
```

Comment:
Timeouts should be picked from configuration objects. But to do that we need access to configuration object which might not available during static initialization.
To solve this, we can move all this initialization to lazy processing. When there is a first call to create bookkeeper client, during that we do this initialization.